client/my-sites/controller.js

```diff
@@ -109,7 +109,7 @@ module.exports = {
 	 * Set up site selection based on last URL param and/or handle no-sites error cases
 	 */
 	siteSelection( context, next ) {
-		const siteID = route.getSiteFragment( context.path );
+		const siteID = context.params.site_id;
```

Comment:
@jordwest my commit message at @2841cb255cbb924dc0b2d4dfe7b10130ac875b4a I guess explains it half way. This change is needed because this pull request introduces a new URL pattern which causes `getSiteFragment` to no longer function as expected. Pulling the `site_id` out of the path context eliminates the problem since its not actually necessary to parse the URL path for it.